### DISCUSSION TITLE
Swift project erroring on iOS builds with "couldn’t be saved in the folder Tests”

### DISCUSSION THREAD

**Discussion by heckj** [pointfreeco/swift-snapshot-testing#602] [Q&A]
I suspect there's a workaround or something that I'm just missing, did some searches on the error message, but haven't spotted anything yet to try.

I'm using the snapshot testing with code in a library, all included within a swift-package. When I'm running on macOS and testing on macOS, it's saving the images as you'd expect into the relevant directory structure. The library itself is meant to run on macOS and iOS, so I'd like to maintain snapshots for both. When I open `Package.swift` in Xcode, it defaults to building and testing on macOS - which works. But if I change the target to an iOS target (building for iPhone 13, for example - or doing the rough equivalent of this through my CI machine), I'm getting the following error:

```
Test Case '-[SnapshotTests.BarChartRenderTests testHorizontalBarChartImageRendering]' started.
/Users/heckj/src/Chart/Tests/SnapshotTests/BarChartRenderTests.swift:38: error: -[SnapshotTests.BarChartRenderTests testHorizontalBarChartImageRendering] : failed - The file “SnapshotTests” couldn’t be saved in the folder “Tests”.
Test Case '-[SnapshotTests.BarChartRenderTests testHorizontalBarChartImageRendering]' failed (0.231 seconds).
Test Case '-[SnapshotTests.BarChartRenderTests testVerticalBarChartImageRendering]' started.
```

When I'm testing through CI (GitHub Actions), I'm doing so through `xcodebuild`:
```
xcodebuild test -scheme 'Chart' -destination 'platform=iOS Simulator,OS=15.2,name=iPhone 8' -showBuildTimingSummary -enableCodeCoverage YES build test
```

(It's pulled down version 1.9.0 of the library, from commit `f8a9c997c3c1dab4e216a8ec9014e23144cbab37`)

Is there any obvious debugging I should be doing to show myself what's causing the permission error?

And is there a preferred technique for managing sets of snapshots built from different platform targets?

Right now I'm working around this by locking the tests to just when run on macOS with `#if os(macOS)`, but I'd very much to get and validate snapshots from iOS as well.

**Comment by kaisegni** [comment]
Any news about this ? I'm running into the same issue :/
